client.go

```diff
@@ -142,24 +147,31 @@ func newClient(zkquorum string, options ...Option) *client {
 		regionReadTimeout:   region.DefaultReadTimeout,
 		done:                make(chan struct{}),
 		newRegionClientFn:   region.NewClient,
+		logger:              slog.Default(),
 	}
 	for _, option := range options {
 		option(c)
 	}
+	c.logger.Info("Creating new admin client.", "Host", slog.StringValue(zkquorum))
 
 	//Have to create the zkClient after the Options have been set
 	//since the zkTimeout could be changed as an option
-	c.zkClient = zk.NewClient(zkquorum, c.zkTimeout, c.zkDialer)
+	c.zkClient = zk.NewClient(zkquorum, c.zkTimeout, c.zkDialer, c.logger)
 
 	return c
 }
 
 // DebugState information about the clients keyRegionCache, and clientRegionCache
-func DebugState(client Client) ([]byte, error) {
+func DebugState(c Client) ([]byte, error) {
 
-	debugInfoJson, err := json.Marshal(client)
+	debugInfoJson, err := json.Marshal(c)
 	if err != nil {
-		log.Errorf("Cannot turn client into JSON bytes array: %v", err)
+		if cclient, ok := c.(*client); ok {
+			cclient.logger.Error("Cannot turn client into JSON bytes array", "error", err)
+		} else {
+			slog.Error("unknown client type", "type", reflect.TypeOf(c))
```

Comment:
Adding reflect just to figure out the type seems wrong

Reply:
I was thinking about maybe it would be helpful to log the type to get more info when having error. Agree that it might be overkill. 
Removed: https://github.com/tsuna/gohbase/pull/257/files#diff-4b667feae66c9d46b21b9ecc19e8958cf4472d162ce0a47ac3e8386af8bbd8cfR170